packages/ketcher-core/src/domain/entities/DrawingEntitiesManager.ts

```diff
@@ -1052,6 +1077,19 @@ export class DrawingEntitiesManager {
     rearrangedMonomersSet.add(nucleotide.sugar.id);
     rearrangedMonomersSet.add(nucleotide.rnaBase?.id);
 
+    const matrixX = isNewRow
+      ? snakeLayoutMatrix.height
+      : snakeLayoutMatrix.height - 1;
+    const matrixY = isNewRow
+      ? 0
+      : snakeLayoutMatrix.getRow(matrixX)?.length ?? 0;
+
+    snakeLayoutMatrix.set(
+      matrixX,
+      matrixY,
+      new Cell(null, [], matrixX, matrixY, nucleotide.sugar),
+    );
```

Comment:
These calculations are performed twice here and above, maybe it makes sense to separate them into some function/class method